api/millenium_patron.py

```diff
@@ -99,6 +106,31 @@ class MilleniumPatronAPI(BasicAuthenticationProvider, XMLParser):
           "default": PIN_AUTHENTICATION_MODE
         }
     ] + BasicAuthenticationProvider.SETTINGS
+
+    LIBRARY_SETTINGS = [
+        { "key": PATRON_FIELD_RESTRICTION_FIELD,
+          "label": _("Patron record restriction field"),
+          "type": "select",
+          "description": _("Field on the patron record used for <i>patron record restriction</i> below."),
```

Comment:
My initial thought was that this setting should be optional since not everyone will use it, but it would be more work to validate that it gets set whenever the patron field restriction setting does, so I think it's okay to set it even when it's not being used.

I would maybe be more explicit in the description that it won't be used unless you set the patron record restriction setting.